src/util/include/util/move_only_function.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Xournal++
+ *
+ * Rudimentary port of C++23's std::move_only_function
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <type_traits>
+#include <utility>
+
+namespace xoj::util {
+template <class T>
+class move_only_function final {};
```

Comment:
Should be only a class template prototype (remove `{}`). Since an invalid specialisation should result in a compile error directly.